util/compression_test.cc

```diff
@@ -0,0 +1,204 @@
+        SyncPoint::GetInstance()->SetCallBack(
+            "AutoSkipCompressorWrapper::CompressBlock::exploitOrExplore",
+            set_exploration);
+      } else {
+        SyncPoint::GetInstance()->SetCallBack(
+            "AutoSkipCompressorWrapper::CompressBlock::exploitOrExplore",
+            unset_exploration);
+      }
+      SyncPoint::GetInstance()->EnableProcessing();
+
+      auto compressed_count = PopStat(NUMBER_BLOCK_COMPRESSED);
+      auto bypassed_count = PopStat(NUMBER_BLOCK_COMPRESSION_BYPASSED);
+      auto rejected_count = PopStat(NUMBER_BLOCK_COMPRESSION_REJECTED);
+      auto total = compressed_count + rejected_count + bypassed_count;
+      int rejection_percentage, bypassed_percentage, compressed_percentage;
+      if (total != 0) {
+        rejection_percentage = static_cast<int>(rejected_count * 100 / total);
+        bypassed_percentage = static_cast<int>(bypassed_count * 100 / total);
+        compressed_percentage =
+            static_cast<int>(compressed_count * 100 / total);
+      }
+      // use mulitple of 10 to get correct assertion
+      switch (multiple_of_10) {
+        case 1:
+          EXPECT_EQ(rejection_percentage, 60);
+          EXPECT_EQ(bypassed_percentage, 0);
+          EXPECT_EQ(compressed_percentage, 40);
+          break;
+        case 2:
+          EXPECT_EQ(rejection_percentage, 0);
+          EXPECT_EQ(bypassed_percentage, 100);
+          EXPECT_EQ(compressed_percentage, 0);
+          break;
+        case 3:
+          EXPECT_EQ(rejection_percentage, 40);
+          EXPECT_EQ(bypassed_percentage, 0);
+          EXPECT_EQ(compressed_percentage, 60);
+          break;
+        case 4:
+          EXPECT_EQ(rejection_percentage, 60);
+          EXPECT_EQ(bypassed_percentage, 0);
+          EXPECT_EQ(compressed_percentage, 40);
+      }
+    }
+    num_keys_++;
+    return true;
+  }
+  uint64_t PopStat(Tickers t) { return statistics_->getAndResetTickerCount(t); }
+
+ private:
+  int window_;
+  int num_keys_;
+  const BlockBuilder& data_block_builder_;
+  std::shared_ptr<Statistics> statistics_;
+};
+
+class AutoSkipTestFlushBlockPolicyFactory : public FlushBlockPolicyFactory {
+ public:
+  explicit AutoSkipTestFlushBlockPolicyFactory(
+      const int window, std::shared_ptr<Statistics> statistics)
+      : window_(window), statistics_(statistics) {}
+
+  virtual const char* Name() const override {
+    return "AutoSkipTestFlushBlockPolicyFactory";
+  }
+
+  virtual FlushBlockPolicy* NewFlushBlockPolicy(
+      const BlockBasedTableOptions& /*table_options*/,
+      const BlockBuilder& data_block_builder) const override {
+    (void)data_block_builder;
+    return new AutoSkipTestFlushBlockPolicy(window_, data_block_builder,
+                                            statistics_);
+  }
+
+ private:
+  int window_;
+  std::shared_ptr<Statistics> statistics_;
+};
+
+class DBAutoSkip : public DBTestBase {
+ public:
+  Options options;
+  Random rnd_;
+  int key_cursor_ = 0;
```

Comment:
initialize key_cursor_ in the constructor DBAutoSkip() to align with other variables.